src/entities/removeLiquidity/index.ts

```diff
@@ -33,6 +34,38 @@ export class RemoveLiquidity implements RemoveLiquidityBase {
         }
     }
 
+    /**
+     * It's not possible to query Remove Liquidity Recovery in the same way as
+     * other remove liquidity kinds, so a separate handler is required for it.
+     * Since it's not affected by fees or anything other than pool balances,
+     * it's possible to calculate amountsOut as proportional amounts.
+     */
+    public queryRemoveLiquidityRecovery(
```

Comment:
Do you think its a bit misleading/confusing calling this `query`? Based off the other methods named this way I would expect it to get an up to date onchain result when called.

Reply:
My reasoning for keeping a similar method name is because even though it uses a different approach to "query", it returns the same results as the other ones. And should be used in the same way: query -> buildCall
Not sure if external users would mind how the "query" is performed, as long as it behaves the same way.

I get that it's using api balances instead of on-chain, so it doesn't necessarily behave the same way, but it should be pretty straightforward to fetch on-chain balances instead of relying on api, which I agree would make more sense. Would that be enough? Or do you think we'd still be better off with another method name?

Reply:
I think the issue I was imagining was a user making a query then waiting some time before making another query to get an updated result. In this case it wouldn't give any update, even if the state had changed. That is different from how I would expect it to work. 
I think as long as it was changed to always use the latest state when query was called then it doesn't matter how that is done.

Reply:
done ✅